paddle/math/Matrix.h

```diff
@@ -370,7 +370,7 @@ class Matrix : public BaseMatrix {
    * allocate matTrans' memory outside, then set memAlloc as false;
    * else set as true.
    */
-  virtual void transpose(MatrixPtr matTrans, bool memAlloc) {
+  virtual void transpose(MatrixPtr& matTrans, bool memAlloc) {
```

Comment:
The transpose() and inverse() were originally designed to operate on "this" and produce the dst. The test case in test_matrixCompare missed the case when memAlloc is true and the memory is allocated inside the function. If you think the implementation can be further improved, then I can go to change it for both transpose() and inverse().